nitro-root-enclave/src/main.rs

```diff
@@ -0,0 +1,467 @@
+            }
+            NitroRootEnclaveMessage::SetMexicoCityHashHack(hash) => {
+                let status = set_mexico_city_hash_hack(hash)?;
+                NitroRootEnclaveMessage::Status(status)
+            }
+            NitroRootEnclaveMessage::NativeAttestation(challenge, device_id) => {
+                println!("nitro-root-enclave::main received NativeAttestaion message");
+                let (proxy_token, public_key) =
+                    native_attestation(&challenge, device_id).map_err(|err| {
+                        format!(
+                            "nitro-root-enclave::main native_attestation failed:{:?}",
+                            err
+                        )
+                    })?;
+                NitroRootEnclaveMessage::TokenData(proxy_token, public_key)
+            }
+            NitroRootEnclaveMessage::ProxyAttestation(challenge, native_token, enclave_name) => {
+                println!("nitro-root-enclave::main received ProxyAttesstation message");
+                let proxy_token = proxy_attestation(&challenge, &native_token, enclave_name)
+                    .map_err(|err| {
+                        println!("proxy_attestation failed");
+                        format!(
+                            "nitro-root-enclave::main proxy_attestation failed:{:?}",
+                            err
+                        )
+                    })?;
+                let device_id: i32 = {
+                    let di_guard = DEVICE_ID.lock().map_err(|err| {
+                        println!("Failed to obtain lock on DEVICE_ID");
+                        format!(
+                            "nitro-root-enclave::main failed to obtain lock on DEVICE_ID:{:?}",
+                            err
+                        )
+                    })?;
+                    match &*di_guard {
+                        Some(did) => *did,
+                        None => {
+                            return Err(format!(
+                                "nitro-root-enclave::main DEVICE_ID is not set up?"
+                            ))
+                        }
+                    }
+                };
+                let device_private_key = {
+                    let dkp_guard = DEVICE_KEY_PAIR.lock()
+                        .map_err(|err| format!("ntiro-root-enclave:proxy_attestation failed to obtain lock on DEVICE_KEY_PAIR:{:?}", err))?;
+                    match &*dkp_guard {
+                        Some((_key, bytes)) => bytes.clone(),
+                        None => {
+                            return Err(format!(
+                                "nitro-root-enclave::proxy_attestation Device Key pair is not populated?"
+                            ))
+                        }
+                    }
+                };
+                let device_public_key = {
+                    // Oddity: the current way of extracting the public key from the private
```

Comment:
A similar comment, and similar code, appears above.  Maybe factor this out into a separate function, so if any better replacement comes along this can be more easily replaced?

Reply:
Issue #65